File: plugins/invite.py
# Cube. Copyright (C) Jake Gealer <jake@gealer.email> 2017-2018.
# This Source Code Form is subject to the terms of the Mozilla Public
# License, v. 2.0. If a copy of the MPL was not distributed with this
# file, You can obtain one at http://mozilla.org/MPL/2.0/.

import discord
# Imports go here.

def Plugin(app):

    @app.command("DM's a invite for this bot.")
    async def invite(app):
        try:
            await app.whisper("https://discordapp.com/oauth2/authorize?client_id={}&scope=bot&permissions=2080898303".format(app.dclient.user.id))
            await app.say(embed=discord.Embed(title="📬 Check DM's"))
        except:
            await app.say(embed=discord.Embed(title="Could not DM.", color=0xff0000))
    # Allows you to invite to another server.

File: cube/commands/invite.go
package commands

import (
	"github.com/jakemakesstuff/Cube/cube/categories"
	"github.com/jakemakesstuff/Cube/cube/command_processor"
	"github.com/jakemakesstuff/Cube/cube/messages"
)

func init() {
	commandprocessor.Commands["invite"] = &commandprocessor.Command{
		Description: "Gives you an invite to the bot.",
		Category:    categories.INFORMATIONAL,
		Function: func(Args *commandprocessor.CommandArgs) {
			InviteURL := "https://discordapp.com/oauth2/authorize?client_id=" + Args.Session.State.User.ID + "&scope=bot&permissions=1546775744"
			messages.GenericText(
				Args.Channel, "Invite:",
				"You can invite the bot [here]("+InviteURL+").",
				nil, Args.Session,
			)
		},
	}
}
